sam.c

```diff
@@ -1282,6 +1282,31 @@ int bam_aux_del(bam1_t *b, uint8_t *s)
     return 0;
 }
 
+int bam_aux_update_str(bam1_t *b, const char tag[2], int len, uint8_t *data)
+{
+    uint8_t *s = bam_aux_get(b,tag);
+    if (!s) return -1;
+    char type = *s;
+    if (type != 'Z') { fprintf(stderr,"bam_aux_update_str() called for type '%c' instead of 'Z'\n", type); abort(); }
+    bam_aux_del(b,s);
+    s -= 2;
+    int l_aux = bam_get_l_aux(b);
+    uint8_t *aux = bam_get_aux(b);
+
+    b->l_data += 3 + len;
+    if (b->m_data < b->l_data) {
+        b->m_data = b->l_data;
+        kroundup32(b->m_data);
+        b->data = (uint8_t*)realloc(b->data, b->m_data);
```

Comment:
This should check if b->data is NULL and fail if so.

*Ideally* it's be something like:

```
uint8_t *tmp = (uint8_t *)realloc(b->data, b->m_data);
if (!tmp) return -1;
b->data = tmp;
```

Otherwise on failure to realloc it's trashed your existing value.